sycl/include/CL/sycl/INTEL/esimd/esimd.hpp

```diff
@@ -251,11 +251,13 @@ template <typename Ty, int N> class simd {
     return LHS;                                                                \
   }
 
-  DEF_LOGIC_OP(&, &=)
-  DEF_LOGIC_OP(|, |=)
-  DEF_LOGIC_OP(^, ^=)
+  DEF_BITWISE_OP(&, &=)
+  DEF_BITWISE_OP(|, |=)
+  DEF_BITWISE_OP(^, ^=)
+  DEF_BITWISE_OP(<<, <<=)
```

Comment:
Looks good. I assume the underlying Clang vector operator will take care of integer promotion as expected.

Looks like the support for % and %= are still missing - could be added to the end of DEF_BINOP list @line 210.